src/jit/lower.cpp

```diff
@@ -809,112 +808,166 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
     isOnStack = info->regNum == REG_STK;
 #endif // !FEATURE_UNIX_AMD64_STRUCT_PASSING
 
-    if (!isOnStack)
+#ifdef _TARGET_ARM_
+    // Struct can be split into register(s) and stack on ARM
+    if (info->isSplit)
     {
-#ifdef FEATURE_SIMD
-        // TYP_SIMD8 is passed in an integer register.  We need the putArg node to be of the int type.
-        if (type == TYP_SIMD8 && genIsValidIntReg(info->regNum))
+        if (arg->OperGet() != GT_OBJ)
         {
-            type = TYP_LONG;
+            NYI_ARM("Lowering: Oper for struct argument is not GT_OBJ");
         }
+
+        putArg = new (comp, GT_PUTARG_SPLIT)
+            GenTreePutArgSplit(arg, info->slotNum PUT_STRUCT_ARG_STK_ONLY_ARG(info->numSlots), info->numRegs,
+                               info->isHfaRegArg, call->IsFastTailCall(), call);
+
+        // Set GC Pointer info
+        BYTE*    gcLayout = new (comp, CMK_Codegen) BYTE[info->numSlots + info->numRegs];
+        unsigned numRefs  = comp->info.compCompHnd->getClassGClayout(arg->gtObj.gtClass, gcLayout);
```

Comment:
If you have a `GT_OBJ` node here, you can get the gc info from the `GenTreeObj` node, rather than having to make another call across the JIT\EE interface. And perhaps you could simply construct the layout for the `GT_FIELD_LIST` case once that is handled.